### ISSUE TITLE
azurerm_postgresql_flexible_server_active_directory_administrator can't be destroyed when linked to Group and active_directory_auth_enabled is on

### ISSUE THREAD

**Issue by mkarbo** [hashicorp/terraform-provider-azurerm#23851]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.4.6

### AzureRM Provider Version

3.75.0

### Affected Resource(s)/Data Source(s)

azurerm_postgresql_flexible_server_active_directory_administrator

### Terraform Configuration Files

```hcl
resource "azurerm_postgresql_flexible_server_active_directory_administrator" "aad_admin" {
  server_name         = azurerm_postgresql_flexible_server.db.name
  resource_group_name = var.resource_group_name
  tenant_id           = var.tenant_id
  object_id           = data.azuread_group.admin.object_id
  principal_name      = data.azuread_group.admin.display_name
  principal_type      = "Group"
}
```

### Debug Output/Panic Output

```shell
module.XXX.azurerm_postgresql_flexible_server_active_directory_administrator.aad_admin: Still destroying... [id=/subscriptions/XXXX...s/XXX, 23m39s elapsed]
```

### Expected Behaviour

It should be destroyed

### Actual Behaviour

It never destroys	

### Steps to Reproduce

Use this (populate with your own variables)
```terraform
resource "azurerm_postgresql_flexible_server" "db" {
  ...
  authentication {
    active_directory_auth_enabled = true
    password_auth_enabled         = false
    tenant_id                     = var.tenant_id
  }

}

data "azuread_group" "admin" {
  object_id        = var.admin_group_id
  security_enabled = true
}

resource "azurerm_postgresql_flexible_server_active_directory_administrator" "aad_admin" {
  server_name         = azurerm_postgresql_flexible_server.db.name
  resource_group_name = var.resource_group_name
  tenant_id           = var.tenant_id
  object_id           = data.azuread_group.admin.object_id
  principal_name      = data.azuread_group.admin.display_name
  principal_type      = "Group"
}
```

And then run `terraform apply`.

### Important Factoids

Azure West europe

### References

https://discuss.hashicorp.com/t/delete-postgresql-flexible-server-with-ad-admins-enabled/57042 seems to be reporting the same thing

**Comment by neil-yechenwei** [comment]
Thanks for raising this issue. Seems azurerm_postgresql_flexible_server_active_directory_administrator is destroyed successfully with latest azurerm provider and below tf config. Could you try below tf config with latest azurerm provider to see if the issue still exists? Thanks.

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/19754191/dc70be1e-e143-4467-97d9-2051c23b1b19)

tf config:
```
provider "azurerm" {
  features {}
}

provider "azuread" {}

data "azurerm_client_config" "current" {}

resource "azurerm_resource_group" "test" {
  name     = "acctestRG-psql-test01"
  location = "westeurope"
}

resource "azurerm_postgresql_flexible_server" "test" {
  name                = "acctest-fs-test01"
  resource_group_name = azurerm_resource_group.test.name
  location            = azurerm_resource_group.test.location
  storage_mb          = 32768
  version             = "12"
  sku_name            = "GP_Standard_D2s_v3"
  zone                = "2"

  authentication {
    active_directory_auth_enabled = true
    password_auth_enabled         = false
    tenant_id                     = data.azurerm_client_config.current.tenant_id
  }
}

data "azuread_group" "test" {
  object_id = "xx-xxx-xxxx-xxxx"
}

resource "azurerm_postgresql_flexible_server_active_directory_administrator" "test" {
  server_name         = azurerm_postgresql_flexible_server.test.name
  resource_group_name = azurerm_resource_group.test.name
  tenant_id           = data.azurerm_client_config.current.tenant_id
  object_id           = data.azuread_group.test.object_id
  principal_name      = data.azuread_group.test.display_name
  principal_type      = "Group"
}
```

**Comment by mkarbo** [comment]
Still no luck

**Comment by jkfran** [comment]
I had this issue and it went away after upgrading `azurerm` to `3.87.0`.

**Comment by jkfran** [comment]
I had this issue again recently, and basically, whenever there is stuff in the DB that belongs to the account it will fail.

I solved it by running first:
```sql
DROP OWNED BY group_name_here;
REVOKE ALL PRIVILEGES ON ALL TABLES IN SCHEMA public FROM group_name_here;
REVOKE ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA public FROM group_name_here;
REVOKE ALL PRIVILEGES ON ALL FUNCTIONS IN SCHEMA public FROM group_name_here;
```

I hope this helps.

**Comment by mkarbo** [comment]
For us, stopping terraform when it get stuck, stopping the DB in the azure portal and then deleting the flexible server manually in the portal, and then continuing _terraform destroy_ worked as well. quite bad for CI though, sadly.

**Comment by feischml** [comment]
> I had this issue again recently, and basically, whenever there is stuff in the DB that belongs to the account it will fail.
> 
> I solved it by running first:
> 
> ```sql
> DROP OWNED BY group_name_here;
> REVOKE ALL PRIVILEGES ON ALL TABLES IN SCHEMA public FROM group_name_here;
> REVOKE ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA public FROM group_name_here;
> REVOKE ALL PRIVILEGES ON ALL FUNCTIONS IN SCHEMA public FROM group_name_here;
> ```
> 
> I hope this helps.

Yes, this was also the solution for me, just that I had access to the DB and directly deleted the group_name from DBeaver.

**Comment by katbyte** [comment]
Thank you for taking the time to report this issue, and apologies that it has sat without a resolution for so long.

This bug was reported against v3.x of the AzureRM provider. The provider has since moved on to v5.x, and the affected functionality has often been significantly reworked across those major releases, so bug reports from legacy major versions prior to v4.0 are generally no longer actionable as written.

As part of a cleanup of historical issues we are closing bug reports that target legacy provider versions. If you are still running into this behaviour on the current major version, please open a new issue with up-to-date reproduction details (feel free to reference this one) so it can be triaged with current context.

Thank you again for the report, and for your patience.
